azurerm/data_source_cosmos_db_account.go

```diff
@@ -0,0 +1,218 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmCosmosDBAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmCosmosDBAccountRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": locationForDataSourceSchema(),
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
```

Comment:
_really minor_ we order fields by Required -> Optional -> Computed; since `name` and `resource_group_name` are both required here and everything else (including `location`) is Computed - it'd be consistent to move these up - but it's not a blocker / an FYI for the future